lib/pg_ha_migrations/relation.rb

```diff
@@ -166,4 +178,37 @@ def valid?
       SQL
     end
   end
+
+  class TableCollection
+    include Enumerable
+
+    attr_reader :raw_set
+
+    delegate :each, to: :raw_set
+    delegate :mode, to: :first, allow_nil: true
```

Comment:
Can there never be different modes on the different tables?

Reply:
With the ways we are currently interacting with this collection, no. I can probably enforce that in the initializer if you'd like to be a bit more defensive. The only downside is more work for me 😆 . Adding a guard like that would require dedicated unit tests (we exercise these classes currently via functional tests of the methods that use them).

Reply:
I think it's probably worth being defensive if it's not that difficult.